### PR TITLE
Zms 576

### PR DESCRIPTION
**Pull request by grishick**
Testing:
`zimbra@zimbra-dev:/Users/gsolovyev/ubuntuhome/git/zm-mailbox$ zmsoap -z RunUnitTestsRequest test=TestUnread ../test=TestFilter ../test=TestBlobDeduper ../test=TestCalDav ../test=TestDbUtil ../test=TestDeployZimlet ../test=TestSpellCheck ../test=TestUnread`

&lt;RunUnitTestsResponse numFailed="0" numSkipped="0" numExecuted="114" xmlns="urn:zimbraAdmin"&gt;
  &lt;results&gt;
    &lt;completed name="testReadMessage1" class="com.zimbra.qa.unittest.TestUnread" execSeconds="0.17"/&gt;
    &lt;completed name="testReadMessage2" class="com.zimbra.qa.unittest.TestUnread" execSeconds="0.10"/&gt;
    &lt;completed name="testReadAllMessages" class="com.zimbra.qa.unittest.TestUnread" execSeconds="0.13"/&gt;
    &lt;completed name="testReadConversation" class="com.zimbra.qa.unittest.TestUnread" execSeconds="0.10"/&gt;
    &lt;completed name="testReadFolder1" class="com.zimbra.qa.unittest.TestUnread" execSeconds="0.11"/&gt;
    &lt;completed name="testReadFolder2" class="com.zimbra.qa.unittest.TestUnread" execSeconds="0.11"/&gt;
    &lt;completed name="testReadAllFolders" class="com.zimbra.qa.unittest.TestUnread" execSeconds="0.12"/&gt;
    &lt;completed name="testReadTag1" class="com.zimbra.qa.unittest.TestUnread" execSeconds="0.12"/&gt;
    &lt;completed name="testReadTag2" class="com.zimbra.qa.unittest.TestUnread" execSeconds="0.11"/&gt;
    &lt;completed name="testMoveMessage" class="com.zimbra.qa.unittest.TestUnread" execSeconds="0.12"/&gt;
    &lt;completed name="testMoveConversation" class="com.zimbra.qa.unittest.TestUnread" execSeconds="0.10"/&gt;
    &lt;completed name="testTagMessage" class="com.zimbra.qa.unittest.TestUnread" execSeconds="0.12"/&gt;
    &lt;completed name="testTagConversation" class="com.zimbra.qa.unittest.TestUnread" execSeconds="0.10"/&gt;
    &lt;completed name="testMoveToTrash" class="com.zimbra.qa.unittest.TestUnread" execSeconds="0.11"/&gt;
    &lt;completed name="testSearch" class="com.zimbra.qa.unittest.TestUnread" execSeconds="0.10"/&gt;
    &lt;completed name="testDeleteConversation" class="com.zimbra.qa.unittest.TestUnread" execSeconds="0.11"/&gt;
    &lt;completed name="testDeleteFolder2" class="com.zimbra.qa.unittest.TestUnread" execSeconds="0.11"/&gt;
    &lt;completed name="testDeleteFolder1" class="com.zimbra.qa.unittest.TestUnread" execSeconds="0.11"/&gt;
    &lt;completed name="testOutgoingFiltersWithSendMsg" class="com.zimbra.qa.unittest.TestFilter" execSeconds="0.42"/&gt;
    &lt;completed name="testCurrentTimeTest" class="com.zimbra.qa.unittest.TestFilter" execSeconds="0.53"/&gt;
    &lt;completed name="testCurrentDayOfWeekTest" class="com.zimbra.qa.unittest.TestFilter" execSeconds="0.53"/&gt;
    &lt;completed name="testAddressTest" class="com.zimbra.qa.unittest.TestFilter" execSeconds="0.57"/&gt;
    &lt;completed name="testAddressTestPart" class="com.zimbra.qa.unittest.TestFilter" execSeconds="0.58"/&gt;
    &lt;completed name="testReplyAction" class="com.zimbra.qa.unittest.TestFilter" execSeconds="0.91"/&gt;
    &lt;completed name="testNotifyAction" class="com.zimbra.qa.unittest.TestFilter" execSeconds="0.88"/&gt;
    &lt;completed name="testNotifyActionUseOrigHeaders" class="com.zimbra.qa.unittest.TestFilter" execSeconds="0.94"/&gt;
    &lt;completed name="testNotifyActionCopyAllOrigHeaders" class="com.zimbra.qa.unittest.TestFilter" execSeconds="1.00"/&gt;
    &lt;completed name="testNotifyWithDiscard" class="com.zimbra.qa.unittest.TestFilter" execSeconds="0.52"/&gt;
    &lt;completed name="testMultipleMultilineText" class="com.zimbra.qa.unittest.TestFilter" execSeconds="0.33"/&gt;
    &lt;completed name="testMarkRead" class="com.zimbra.qa.unittest.TestFilter" execSeconds="0.52"/&gt;
    &lt;completed name="testHeaderFolding" class="com.zimbra.qa.unittest.TestFilter" execSeconds="0.46"/&gt;
    &lt;completed name="testInvite" class="com.zimbra.qa.unittest.TestFilter" execSeconds="2.64"/&gt;
    &lt;completed name="testManyAsterisks" class="com.zimbra.qa.unittest.TestFilter" execSeconds="0.32"/&gt;
    &lt;completed name="testDateFiltering" class="com.zimbra.qa.unittest.TestFilter" execSeconds="0.84"/&gt;
    &lt;completed name="testHeaderMatches" class="com.zimbra.qa.unittest.TestFilter" execSeconds="0.42"/&gt;
    &lt;completed name="testBodyContains" class="com.zimbra.qa.unittest.TestFilter" execSeconds="1.98"/&gt;
    &lt;completed name="testFullMatchAfterPartialMatch" class="com.zimbra.qa.unittest.TestFilter" execSeconds="0.42"/&gt;
    &lt;completed name="testSpecialCharInBody" class="com.zimbra.qa.unittest.TestFilter" execSeconds="0.40"/&gt;
    &lt;completed name="testRedirect" class="com.zimbra.qa.unittest.TestFilter" execSeconds="1.50"/&gt;
    &lt;completed name="testRedirectMailLoop" class="com.zimbra.qa.unittest.TestFilter" execSeconds="0.87"/&gt;
    &lt;completed name="testAttachment" class="com.zimbra.qa.unittest.TestFilter" execSeconds="0.59"/&gt;
    &lt;completed name="testParseSize" class="com.zimbra.qa.unittest.TestFilter" execSeconds="0.27"/&gt;
    &lt;completed name="testStripBracketsAndQuotes" class="com.zimbra.qa.unittest.TestFilter" execSeconds="0.28"/&gt;
    &lt;completed name="testPositiveAndNegative" class="com.zimbra.qa.unittest.TestFilter" execSeconds="0.30"/&gt;
    &lt;completed name="testBlobDeduper" class="com.zimbra.qa.unittest.TestBlobDeduper" execSeconds="2.13"/&gt;
    &lt;completed name="testBadBasicAuth" class="com.zimbra.qa.unittest.TestCalDav" execSeconds="0.09"/&gt;
    &lt;completed name="testPostToSchedulingOutbox" class="com.zimbra.qa.unittest.TestCalDav" execSeconds="0.46"/&gt;
    &lt;completed name="testBadPostToSchedulingOutbox" class="com.zimbra.qa.unittest.TestCalDav" execSeconds="0.24"/&gt;
    &lt;completed name="testCalendarQueryOnInbox" class="com.zimbra.qa.unittest.TestCalDav" execSeconds="0.83"/&gt;
    &lt;completed name="testCalendarQueryOnOutbox" class="com.zimbra.qa.unittest.TestCalDav" execSeconds="0.22"/&gt;
    &lt;completed name="testPropFindSupportedReportSetOnInbox" class="com.zimbra.qa.unittest.TestCalDav" execSeconds="0.10"/&gt;
    &lt;completed name="testPropFindSupportedReportSetOnOutbox" class="com.zimbra.qa.unittest.TestCalDav" execSeconds="0.10"/&gt;
    &lt;completed name="testPropFindSupportedCalendarComponentSetOnInbox" class="com.zimbra.qa.unittest.TestCalDav" execSeconds="0.10"/&gt;
    &lt;completed name="testPropFindSupportedCalendarComponentSetOnOutbox" class="com.zimbra.qa.unittest.TestCalDav" execSeconds="0.12"/&gt;
    &lt;completed name="testPropFindSupportedCalendarComponentSetOnCalendar" class="com.zimbra.qa.unittest.TestCalDav" execSeconds="0.10"/&gt;
    &lt;completed name="testPropFindSupportedCalendarComponentSetOnTasks" class="com.zimbra.qa.unittest.TestCalDav" execSeconds="0.10"/&gt;
    &lt;completed name="testCreateUsingClientChosenName" class="com.zimbra.qa.unittest.TestCalDav" execSeconds="0.09"/&gt;
    &lt;completed name="testCreateModifyDeleteAttendeeModifyAndCancel" class="com.zimbra.qa.unittest.TestCalDav" execSeconds="2.51"/&gt;
    &lt;completed name="testAndroidMeetingSeries" class="com.zimbra.qa.unittest.TestCalDav" execSeconds="0.73"/&gt;
    &lt;completed name="testSimpleMkcol" class="com.zimbra.qa.unittest.TestCalDav" execSeconds="0.12"/&gt;
    &lt;completed name="testMkcol4addressBook" class="com.zimbra.qa.unittest.TestCalDav" execSeconds="0.10"/&gt;
    &lt;completed name="testPropPatchCalendarFreeBusySetSettingUsingEscapedUrls" class="com.zimbra.qa.unittest.TestCalDav" execSeconds="0.21"/&gt;
    &lt;completed name="testFuzzyTimeZoneMatchGMT_06" class="com.zimbra.qa.unittest.TestCalDav" execSeconds="0.02"/&gt;
    &lt;completed name="testFuzzyTimeZoneMatchGMT_08" class="com.zimbra.qa.unittest.TestCalDav" execSeconds="0.02"/&gt;
    &lt;completed name="testLondonTimeZoneCalledGMTkeepSameName" class="com.zimbra.qa.unittest.TestCalDav" execSeconds="0.02"/&gt;
    &lt;completed name="testLondonTimeZoneCalledGMT" class="com.zimbra.qa.unittest.TestCalDav" execSeconds="0.02"/&gt;
    &lt;completed name="testAttendeeAutoDecline" class="com.zimbra.qa.unittest.TestCalDav" execSeconds="1.20"/&gt;
    &lt;completed name="testAttendeeSuppressedAutoDecline" class="com.zimbra.qa.unittest.TestCalDav" execSeconds="2.75"/&gt;
    &lt;completed name="testCreateContactWithIfNoneMatchTesting" class="com.zimbra.qa.unittest.TestCalDav" execSeconds="0.12"/&gt;
    &lt;completed name="testAppleStyleGroup" class="com.zimbra.qa.unittest.TestCalDav" execSeconds="0.27"/&gt;
    &lt;completed name="testXBusyMacAttach" class="com.zimbra.qa.unittest.TestCalDav" execSeconds="0.12"/&gt;
    &lt;completed name="testAppleCaldavProxyFunctions" class="com.zimbra.qa.unittest.TestCalDav" execSeconds="0.98"/&gt;
    &lt;completed name="testNormalizeSql" class="com.zimbra.qa.unittest.TestDbUtil" execSeconds="0.11"/&gt;
    &lt;completed name="testDatabaseExists" class="com.zimbra.qa.unittest.TestDbUtil" execSeconds="0.04"/&gt;
    &lt;completed name="testValidZimlet" class="com.zimbra.qa.unittest.TestDeployZimlet" execSeconds="0.69"/&gt;
    &lt;completed name="testEmptyAid" class="com.zimbra.qa.unittest.TestDeployZimlet" execSeconds="0.03"/&gt;
    &lt;completed name="testNoAid" class="com.zimbra.qa.unittest.TestDeployZimlet" execSeconds="0.02"/&gt;
    &lt;completed name="testBadAid" class="com.zimbra.qa.unittest.TestDeployZimlet" execSeconds="0.01"/&gt;
    &lt;completed name="testInvalidAction" class="com.zimbra.qa.unittest.TestDeployZimlet" execSeconds="0.06"/&gt;
    &lt;completed name="testBadZimletName" class="com.zimbra.qa.unittest.TestDeployZimlet" execSeconds="0.08"/&gt;
    &lt;completed name="testValidAdminExtension" class="com.zimbra.qa.unittest.TestDeployZimlet" execSeconds="0.07"/&gt;
    &lt;completed name="testAdminExtensionDelegatedAdmin" class="com.zimbra.qa.unittest.TestDeployZimlet" execSeconds="0.06"/&gt;
    &lt;completed name="testUndeployBadName" class="com.zimbra.qa.unittest.TestDeployZimlet" execSeconds="0.02"/&gt;
    &lt;completed name="testZimletDelegatedAdmin" class="com.zimbra.qa.unittest.TestDeployZimlet" execSeconds="0.06"/&gt;
    &lt;completed name="testZipWithTraversal" class="com.zimbra.qa.unittest.TestDeployZimlet" execSeconds="0.02"/&gt;
    &lt;completed name="testZipWithInvalidCharacter" class="com.zimbra.qa.unittest.TestDeployZimlet" execSeconds="0.02"/&gt;
    &lt;completed name="testZipWithAbsolutePath" class="com.zimbra.qa.unittest.TestDeployZimlet" execSeconds="0.06"/&gt;
    &lt;completed name="testPhilsZip" class="com.zimbra.qa.unittest.TestDeployZimlet" execSeconds="0.06"/&gt;
    &lt;completed name="testCheckSpelling" class="com.zimbra.qa.unittest.TestSpellCheck" execSeconds="0.18"/&gt;
    &lt;completed name="testIgnore" class="com.zimbra.qa.unittest.TestSpellCheck" execSeconds="0.12"/&gt;
    &lt;completed name="testGetDictionaries" class="com.zimbra.qa.unittest.TestSpellCheck" execSeconds="0.11"/&gt;
    &lt;completed name="testUnexpectedCharacters" class="com.zimbra.qa.unittest.TestSpellCheck" execSeconds="0.10"/&gt;
    &lt;completed name="testReceiveSpanish" class="com.zimbra.qa.unittest.TestSpellCheck" execSeconds="0.13"/&gt;
    &lt;completed name="testSendSpanish" class="com.zimbra.qa.unittest.TestSpellCheck" execSeconds="0.12"/&gt;
    &lt;completed name="testRussian" class="com.zimbra.qa.unittest.TestSpellCheck" execSeconds="0.13"/&gt;
    &lt;completed name="testAllCaps" class="com.zimbra.qa.unittest.TestSpellCheck" execSeconds="0.11"/&gt;
    &lt;completed name="testReadMessage1" class="com.zimbra.qa.unittest.TestUnread" execSeconds="0.11"/&gt;
    &lt;completed name="testReadMessage2" class="com.zimbra.qa.unittest.TestUnread" execSeconds="0.10"/&gt;
    &lt;completed name="testReadAllMessages" class="com.zimbra.qa.unittest.TestUnread" execSeconds="0.13"/&gt;
    &lt;completed name="testReadConversation" class="com.zimbra.qa.unittest.TestUnread" execSeconds="0.09"/&gt;
    &lt;completed name="testReadFolder1" class="com.zimbra.qa.unittest.TestUnread" execSeconds="0.09"/&gt;
    &lt;completed name="testReadFolder2" class="com.zimbra.qa.unittest.TestUnread" execSeconds="0.09"/&gt;
    &lt;completed name="testReadAllFolders" class="com.zimbra.qa.unittest.TestUnread" execSeconds="0.12"/&gt;
    &lt;completed name="testReadTag1" class="com.zimbra.qa.unittest.TestUnread" execSeconds="0.11"/&gt;
    &lt;completed name="testReadTag2" class="com.zimbra.qa.unittest.TestUnread" execSeconds="0.11"/&gt;
    &lt;completed name="testMoveMessage" class="com.zimbra.qa.unittest.TestUnread" execSeconds="0.11"/&gt;
    &lt;completed name="testMoveConversation" class="com.zimbra.qa.unittest.TestUnread" execSeconds="0.12"/&gt;
    &lt;completed name="testTagMessage" class="com.zimbra.qa.unittest.TestUnread" execSeconds="0.11"/&gt;
    &lt;completed name="testTagConversation" class="com.zimbra.qa.unittest.TestUnread" execSeconds="0.10"/&gt;
    &lt;completed name="testMoveToTrash" class="com.zimbra.qa.unittest.TestUnread" execSeconds="0.11"/&gt;
    &lt;completed name="testSearch" class="com.zimbra.qa.unittest.TestUnread" execSeconds="0.10"/&gt;
    &lt;completed name="testDeleteConversation" class="com.zimbra.qa.unittest.TestUnread" execSeconds="0.09"/&gt;
    &lt;completed name="testDeleteFolder2" class="com.zimbra.qa.unittest.TestUnread" execSeconds="0.11"/&gt;
    &lt;completed name="testDeleteFolder1" class="com.zimbra.qa.unittest.TestUnread" execSeconds="0.16"/&gt;
  &lt;/results&gt;
&lt;/RunUnitTestsResponse&gt;

### CODE REVIEW

File: store/src/java/com/zimbra/qa/unittest/TestDeployZimlet.java [hunk 1]
```diff
@@ -48,7 +48,7 @@
 import com.zimbra.soap.admin.type.AttachmentIdAttrib;
 
 /**
- * copy test data from data/unittest to /opt/zimbra/data/unittest before running this test 
+ * copy test data from data/unittest to /opt/zimbra/unittest before running this test
```
Comment (fatkudu) [inline store/src/java/com/zimbra/qa/unittest/TestDeployZimlet.java]: Is there an ant target to do this?

Comment (grishick) [inline store/src/java/com/zimbra/qa/unittest/TestDeployZimlet.java]: was thinking the same thing. I hesitate creating an ANT target for this yet, but we will need it eventually.

**Comment by grishick** [comment]
looks like I'll need to rebase to account for Gren's changes to TestUnread or back out my changes to TestUnread before merging

File: store/src/java/com/zimbra/qa/unittest/TestUnread.java [hunk 2]
```diff
@@ -56,11 +54,11 @@
     private static String TEST_NAME = "TestUnread";
     private static String USER_NAME = TEST_NAME.toLowerCase() + "user1";
 
-    private static String FOLDER1_NAME = TEST_NAME + " Folder 1";
-    private static String FOLDER2_NAME = TEST_NAME + " Folder 2";
-    private static String TAG1_NAME = TEST_NAME + " Tag 1";
-    private static String TAG2_NAME = TEST_NAME + " Tag 2";
-    private static String TAG3_NAME = TEST_NAME + " Tag 3";
+    private static String FOLDER1_NAME =  "Folder 1";
```
Comment (grishick) [inline store/src/java/com/zimbra/qa/unittest/TestUnread.java]: I don't think there is a need to prefix folder names when the account is unique for this test.

**Comment by grishick** [comment]
I've rebased this on develop, so @fatkudu 's changes for TestUnread and TestCalDav are not lost.

**Pull request merged**